motionplan/utils.go

```diff
@@ -1,13 +1,57 @@
 package motionplan
 
 import (
+	"errors"
+	"fmt"
 	"math"
 
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/spatialmath"
 	"go.viam.com/rdk/utils"
 )
 
+// FrameStepsFromRobotPath is a helper function which will extract the waypoints of a single frame from the map output of a robot path.
+func FrameStepsFromRobotPath(frameName string, path []map[string][]referenceframe.Input) ([][]referenceframe.Input, error) {
+	solution := make([][]referenceframe.Input, 0, len(path))
+	for _, step := range path {
+		frameStep, ok := step[frameName]
+		if !ok {
+			return nil, fmt.Errorf("frame named %s not found in solved motion path", frameName)
+		}
+		solution = append(solution, frameStep)
+	}
+	return solution, nil
+}
+
+// EvaluatePlan assigns a numeric score to a plan that corresponds to the cumulative distance between input waypoints in the plan.
+func EvaluatePlan(plan planReturn, planOpts *plannerOptions) (totalCost float64) {
```

Comment:
could be passing in `evaluateFn Constraint` here instead of `planOpts` so we don't need to construct a whole planOpts to use this

Reply:
Anything calling this realistically will have a whole `planOpts` already anyway.

Honestly this just shouldn't be public, since it uses private structs as its params.

Reply:
It could be a useful thing to expose though.  If we make this `EvaluatePlan([][]referenceframe.Input, Constraint)` it would be useful from things like the ompl-evaluation repo

Reply:
Ah, good point. I'll do that.